tests/e2e/vmservice_vm.go

```diff
@@ -615,7 +625,264 @@ var _ bool = ginkgo.Describe("[vmsvc] vm service with csi vol tests", func() {
+		}()
+
+		ginkgo.By("Creating loadbalancing service for ssh with the VM")
+		vmlbsvc := createService4Vm(ctx, vmopC, namespace, vm.Name)
+		defer func() {
+			ginkgo.By("Deleting loadbalancing service for ssh with the VM")
+			err = vmopC.Delete(ctx, &vmopv1.VirtualMachineService{ObjectMeta: metav1.ObjectMeta{
+				Name:      vmlbsvc.Name,
+				Namespace: namespace,
+			}})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Wait for VM to come up and get an IP")
+		vmIp, err := waitNgetVmsvcVmIp(ctx, vmopC, namespace, vm.Name)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Wait and verify PVCs are attached to the VM")
+		gomega.Expect(waitNverifyPvcsAreAttachedToVmsvcVm(ctx, vmopC, cnsopC, vm,
+			[]*v1.PersistentVolumeClaim{pvc})).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify PVCs are accessible to the VM")
+		vm, err = getVmsvcVM(ctx, vmopC, vm.Namespace, vm.Name) // refresh vm info
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for i, vol := range vm.Status.Volumes {
+			_ = formatNVerifyPvcIsAccessible(vol.DiskUuid, i+1, vmIp)
+		}
+
+	})
+
+	/*
+	   VM and PVC both belong to different zones (creation and post creation)
+	   Steps:
+	   1	Assign a zonal spbm policy to test namespace with sufficient quota
+	   2	Create two PVCs say pvc1, pvc2 under zone2
+	   3	Create a VMservice VM say vm1 under zone1 with pvc1
+	   4	verify that vm1 does not come up
+	   5	remove pvc1 from vm1 spec
+	   6	Verify that the vm1 comes up now
+	   7	Attach pvc2 to vm1 and verify that the volume attachment fails
+	   8	Delete vm1
+	   9	delete pvc1, pvc2
+	   10	Remove spbm policy attached to test namespace in step1
+	*/
+	ginkgo.It("[vmsvc-stretched] VM and PVC both belong to same zone", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		scParameters := make(map[string]string)
+		topologyHaMap := GetAndExpectStringEnvVar(topologyHaMap)
+		allowedTopos := createAllowedTopolgies(topologyHaMap, tkgshaTopologyLevels)
+		allowedTopologyHAMap := createAllowedTopologiesMap(allowedTopos)
+		zonalPolicy := GetAndExpectStringEnvVar(envZonalStoragePolicyName)
+
+		pvcAnnotations := make(map[string]string)
+		topoList := []string{}
+		zones := []string{}
+		rand.NewSource(time.Now().UnixNano())
+		for key, val := range allowedTopologyHAMap {
+			for _, topoVal := range val {
+				str := `{"` + key + `":"` + topoVal + `"}`
+				topoList = append(topoList, str)
+				zones = append(zones, topoVal)
+			}
+		}
+		framework.Logf("topoList: %v", topoList)
+		annotationVal := "[" + topoList[0] + "]" // just selecting one topology
+		pvcAnnotations[tkgHARequestedAnnotationKey] = annotationVal
+		framework.Logf("annotationVal :%s, pvcAnnotations: %v", annotationVal, pvcAnnotations)
+
+		ginkgo.By("Creating Pvc with Immediate topology storageclass")
+		createResourceQuota(client, namespace, rqLimit, zonalPolicy)
+		scParameters[svStorageClassName] = zonalPolicy
```

Comment:
This is not required for WCP, we can remove these lines.

Reply:
yes correct, removing them